definitions/features/sync_plans.rb

```diff
@@ -33,12 +33,24 @@ def sync_plan_ids_by_status(enabled = true, filter_ids = nil)
     feature(:foreman_database).query(query).map { |r| r['id'].to_i }
   end
 
+  def validate_sync_plan_ids(ids)
+    ids_condition = ids.map { |id| "'#{id}'" }.join(',')
+    query = <<-SQL
+      SELECT id FROM katello_sync_plans WHERE id IN (#{ids_condition})
+    SQL
+    feature(:foreman_database).query(query).map { |r| r['id'].to_i }
+  end
+
   def make_disable(ids)
     update_records(ids, false)
   end
 
   def make_enable
-    update_records(data[:disabled], true)
+    # remove ids of sync plans which no longer exist in DB
```

Comment:
Shouldn't we tell users to remove IDs from their config?

Reply:
Is it a "config" or just the internal "brain" of the foreman-maintain which shouldn't be adapted by a user itself?

Reply:
data.yml is purely an internal thing that is used for storing things between runs.